crates/storage/provider/src/providers/blockchain_provider.rs

```diff
@@ -1518,6 +1549,23 @@ mod tests {
         };
         provider.canonical_in_memory_state.update_chain(chain);
 
+        // Get canonical, safe, and finalized blocks
+        let canonical_block = blocks.latest().unwrap().clone();
+        let safe_block = blocks.safe().unwrap().clone();
+        let finalized_block = blocks.finalized().unwrap().clone();
```

Comment:
no need to have a separate struct for this, let's just get the block by index from the vector

Reply:
Idea was that a couple of tests also need to access the latest, safe and finalized so I was repeating this indexing quite a lot. Will remove if you prefer index

Reply:
hmm, i see it only in one test